### ISSUE TITLE
assertNotIsInstance's error message is humorously incorrect

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3400]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#3400|
|Type|defect|
|Created|2008-08-28 02:28:19Z|

This test module:

```
#!py
from twisted.trial import unittest

class A(object):                                                                                                                                                                      
    "A"

class AisATest(unittest.TestCase):                                                                                                                                                    
    def test_AisA(self):                                                                                                                                                              
        a = A()
        self.assertNotIsInstance(a, A)
```

Results in this error message:

```
test_notisinstance.AisATest.test_AisA ... Traceback: <class 'twisted.trial.unittest.FailTest'>: <test_notisinstance.A object at 0x84f1dcc> is not an instance of <class 'test_notisinstance.A'>
```

but it seems quite clear that the problem is that an `A` object _is_ an instance of the class `A`.

Attachments:

* [3400.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/881/881cd002ab0b4dc9dab6a131ca7b162f75e77ec9/376e3ae3022464b0fc62f2d0023cda80e3aa53a9.diff) (1217 bytes) - added by therve on 2008-09-01 13:26:01Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3400 3400
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1219890499000000 1219890499000000
changetime__1220276458000000 1220276458000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Fixing markup

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

Fixed in the attached patch.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @therve|
|-|-|

Needs a test.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve removed owner|
|-|-|

Done!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @therve|
|-|-|

Looks good.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set status to `closed`|
|-|-|

(In [24662]) Fix assertNotIsInstance error message when class *is* an instance.

Author: therve
Reviewer: glyph
Fixes [#3400](https://github.com/twisted/twisted/issues/3400)
